src/zenml/config/source.py

```diff
@@ -148,7 +174,8 @@ class DistributionPackageSource(Source):
     version: Optional[str] = None
     type: SourceType = SourceType.DISTRIBUTION_PACKAGE
 
-    @validator("type")
+    @field_validator("type")
+    @classmethod
```

Comment:
Why doesn't Pydantic auto-wrap it into a classmethod?! Not a question to you, for sure 🙂